### PR TITLE
feat: Add label and tooltip for the color schemes control

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR adds the label name in the color schemes control to make it easier for the users to find their color schemes.
It also adds a tooltip when the color scheme label or the color list gets truncated.

### BEFORE

Screenshot 2022-08-10 at 22 05 46

### AFTER

Screenshot 2022-08-10 at 22 02 09

### TESTING INSTRUCTIONS
1. Open Explore
2. Scroll the available color schemes
3. If the label and/or the colors are truncated a tooltip should appear
4. If the label and the colors are fully visible, the tooltip should not appear

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
@kasiazjc it looks like the names AND the palette colors are being truncated. Should that happen to both, or should one of them "win"?

**Comment by geido** [comment]
> @kasiazjc it looks like the names AND the palette colors are being truncated. Should that happen to both, or should one of them "win"?

The proposed design truncates both. I believe it makes sense since you can have a very large amount of colors or a very lenghty name for the palette.

File: superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx [hunk 1]
```diff
@@ -196,7 +179,7 @@ export default class ColorSchemeControl extends React.PureComponent {
     };
 
     return (
-      <Select
+      <StyledSelect
```
Comment (codyml) [inline superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx:182]: I'm noticing a weird thing where the selected color starts at the top of the list, but then moves to another location if you move the mouse outside the list and back in.  I don't see how this PR would have caused that problem, but if it's easy to fix maybe we could fix it here?

https://user-images.githubusercontent.com/13007381/184456288-dd7dcf96-c3ed-4140-b891-5658cfa6b2fb.mov

**Review by codyml — APPROVED** [review]
Code looks good and it works!  Left two comments that are not blocking.  The double truncation does strike me as kind of strange: the row of swatches starts at a different place for each item so the menu looks a little chaotic, and as a user I think my first impression of seeing two ellipses in one menu item would be that there's a layout bug.  I can't think of a perfect alternative, but what about if we made it so we had a max of like 10 swatches and they were right-justified without an ellipsis, and then the title is left-justified, truncated with an ellipsis always at the same point?  We wouldn't see the max amount of information possible but it would look a lot cleaner and the tooltips would show the full content.

Screen Shot 2022-08-12 at 5 43 57 PM

But, also non-blocking if we want to stick with this design.

**Review by jinghua-qa — APPROVED** [review]
LGTM

**Comment by kasiazjc** [comment]
> Code looks good and it works! Left two comments that are not blocking. The double truncation does strike me as kind of strange: the row of swatches starts at a different place for each item so the menu looks a little chaotic, and as a user I think my first impression of seeing two ellipses in one menu item would be that there's a layout bug. I can't think of a perfect alternative, but what about if we made it so we had a max of like 10 swatches and they were right-justified without an ellipsis, and then the title is left-justified, truncated with an ellipsis always at the same point? We wouldn't see the max amount of information possible but it would look a lot cleaner and the tooltips would show the full content.
> 
> Screen Shot 2022-08-12 at 5 43 57 PM
> 
> But, also non-blocking if we want to stick with this design.

Thanks for the feedback Cody! I think actually that is a good idea. We still need an indicator that not all of the colors are shown and I'm thinking - maybe let's add a counter? So, as in the categorical color palette, max number of colors is 12, let's use it's a max shown. It will be the fixed with for the color swatches section. Then between text and colors there is 8px fixed spacing and the rest is title. Does it make sense? All of the things would cut off in the same space and it would declutter the whole thing a little bit. What do you think @geido 

image

File: superset-frontend/src/explore/components/controls/ColorSchemeControl/ColorSchemeLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    </>
+  );
+
+  return (
+    <Tooltip
+      data-testid="tooltip"
+      overlayClassName="color-scheme-tooltip"
+      title={tooltipContent}
+      key={id}
+      visible={showTooltip}
```
[the remark this answers is not in the document]
Comment (geido) [reply superset-frontend/src/explore/components/controls/ColorSchemeControl/ColorSchemeLabel.tsx:85]: Yes this was my first approach as well but unfortunately it won't work as it cannot evaluate the width after mount properly

File: superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx (hunk 1 above)
Comment (geido) [reply superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx:182]: This is a Select component issue that it makes sense to fix in a separate PR. I'll open an issue for this problem. Thanks for catching it!

**Comment by geido** [comment]
> Thanks for the feedback Cody! I think actually that is a good idea. We still need an indicator that not all of the colors are shown and I'm thinking - maybe let's add a counter? So, as in the categorical color palette, max number of colors is 12, let's use it's a max shown. It will be the fixed with for the color swatches section. Then between text and colors there is 8px fixed spacing and the rest is title. Does it make sense? All of the things would cut off in the same space and it would declutter the whole thing a little bit. What do you think @geido
> 
> image

@kasiazjc the two elements cannot be located at the opposite ends as the container can be dragged and expanded. This would create a potential huge space between the label and the colors.

In order to keep this flexible we also should not set a maximum number of 12 colors as the dropdown should behave based on the available horizontal space.

As for changing from ellipsis to a number of hidden colors, that would require a bigger refactor and if we think it is critical, we should do explore it more and I can open another PR.

In the meantime, I made some enhancements so that the options do not look disordered and all consume the same space, which should satisfy what @codyml was asking above. These latest changes truncates both the labels and colors less often, which should improve the feelings about double truncation.

Screenshot 2022-08-17 at 18 38 17

**Comment by kasiazjc** [comment]
> > Thanks for the feedback Cody! I think actually that is a good idea. We still need an indicator that not all of the colors are shown and I'm thinking - maybe let's add a counter? So, as in the categorical color palette, max number of colors is 12, let's use it's a max shown. It will be the fixed with for the color swatches section. Then between text and colors there is 8px fixed spacing and the rest is title. Does it make sense? All of the things would cut off in the same space and it would declutter the whole thing a little bit. What do you think @geido
> > image
> 
> @kasiazjc the two elements cannot be located at the opposite ends as the container can be dragged and expanded. This would create a potential huge space between the label and the colors.
> 
> In order to keep this flexible we also should not set a maximum number of 12 colors as the dropdown should behave based on the available horizontal space.
> 
> As for changing from ellipsis to a number of hidden colors, that would require a bigger refactor and if we think it is critical we should discuss that option and I can open another PR.
> 
> In the meantime, I made some enhancements so that the options do not look disordered and all consume the same space now, which should satisfy what @codyml was asking above. Also, These latest changes truncates both the labels and colors less often, which should improve the feelings about double truncation.
> 
> Screenshot 2022-08-17 at 18 38 17

Oh, I understand. I actually think it looks good! I am totally fine with merging this version. Thank you Diego for working on it 🙏

**Pull request merged**